spark/core.py

```diff
@@ -303,19 +316,45 @@ def fill_arc(self, *args):
 
     def stroke_arc(self, *args):
         self.canvas.stroke_arc(*args)
-    
-    def fill_text(self, *args):
-        self.canvas.font = "{px}px sans-serif".format(px = args[4])
-        self.canvas.fill_text(args[0:3])
-        self.canvas.font = "12px sans-serif"
 
-    def stroke_text(self, *args):
-        self.canvas.font = "{px}px sans-serif".format(px = args[4])
-        self.canvas.stroke_text(args[0:3])
-        self.canvas.font = "12px sans-serif"
+    def text_size(self, *args):
+        if len(args) != 1:
+            raise TypeError(f"text_size expected 1 argument, got {len(args)}")
+        
+        size = args[0]
+        if type(size) == type(str()):
+            if size.endswith('px'):
+                size = size[:-2] # Remove 'px' from end.
+            else:
+                raise TypeError(f"text_size expects a number or a string with a number followed by 'px', got {repr(args[0])}")
+
+        try:
+            size = float(size)
+        except ValueError:
+            raise TypeError(f"text_size expects a number or a string with a number followed by 'px', got {repr(args[0])}")
```

Comment:
In my opinion, I'd recommend simply allowing numbers and disallowing strings. To me, the error message is a bit confusing, and the benefit gained from allowing "px" isn't that big for beginners. You can consider using the helper function `check_type_is_num` instead of manually testing and raising type errors yourself.

```py
self.check_type_is_num("text_size", size)
self.font_settings['size'] = size
```